### PR TITLE
Save session button

### LINKED ISSUE #1985: Implement no-code git Save button as defined
Is this ready to go after https://github.com/SwissDataScienceCenter/renku-ui/issues/1957 is done?

### PR DESCRIPTION
**Pull request by ciyer**
Provide a button save sessions.

## Testing scenarios

- [ ] Anonymous session (save should inform that user needs to log in and re-do their changes)
- [ ] Access level < Developer (save should inform that the user needs to fork the project and push their changes to the fork)
- [ ] No changes (save should inform the user that there are no changes to save)
- [ ] Behind origin (save should inform the user that they need to pull)
- [ ] Ahead of origin (save should let the user save, without a commit message)
- [ ] Dirty (save should provide a dialog box to enter a commit message)

## Screenshots

**Anonymous**
image

**Access level < Developer**
image

**No changes**
image

**Behind origin**
image

**Ahead of origin**
image

**Dirty**
image

Fix #1985 

/deploy renku-notebooks=master #persist

### CODE REVIEW

**Comment by ciyer** [comment]
Add as a comment to avoid causing the branch to re-deploy:

Fix #1985 

## Testing scenarios

- [ ] Anonymous session (save should inform that user needs to log in and re-do their changes)
- [ ] Access level < Developer (save should inform that the user needs to fork the project and push their changes to the fork)
- [ ] No changes (save should inform the user that there are no changes to save)
- [ ] Normal (save should provide a dialog box to enter a commit message)

**Review by andre-code — COMMENTED** [review]
The code looks great! 👏
I see you create the services/session folder for the api, would it make sense to use the feature folder to keep all the APIs in the same place?

I've added a small suggestion regarding the save message in the modal stop session.
It would also be great to add e2e tests for all the cases implemented here.

File: client/src/notebooks/components/StopSession.tsx [hunk 1]
```diff
@@ -55,7 +55,7 @@ function StopSessionModal({ stopSession, stopping, closeModal, isOpen }: StopSes
     <p>
       Are you sure you want to stop this session?
       Make sure to commit any changes you want to be saved for next time
-      by running <code>renku save</code> or using Git.
+      (for example, by using the <b>Save</b> button).
```
[the remark this answers is not in the document]
Comment (ciyer) [reply client/src/notebooks/components/StopSession.tsx]: Good suggestion. I have added this.

**Comment by ciyer** [comment]
Thanks for the review. You are ab

> The code looks great! 👏 I see you create the services/session folder for the api, would it make sense to use the feature folder to keep all the APIs in the same place?
> 
> I've added a small suggestion regarding the save message in the modal stop session. It would also be great to add e2e tests for all the cases implemented here.

Thanks for the suggestions. I have made the changes to the code you suggest.

You are right that there should be e2e tests for this! Given the situation, though, I suggest we merge this without the e2e tests. #1986 will also interact with the sidecar, and I have added the requirement to create e2e tests in the description of that issue.

**Review by andre-code — APPROVED** [review]
lgtm

**Pull request merged**